sentry/src/main/java/io/sentry/transport/RateLimiter.java

```diff
@@ -134,10 +135,29 @@ public boolean isAnyRateLimitActive() {
    *
    * @param hint the Hints
    * @param retry if event should be retried or not
+   * @param envelope the envelope
    */
-  private static void markHintWhenSendingFailed(final @NotNull Hint hint, final boolean retry) {
+  private void markHintWhenSendingFailed(
+      final @NotNull Hint hint, final boolean retry, @NotNull SentryEnvelope envelope) {
     HintUtils.runIfHasType(hint, SubmissionResult.class, result -> result.setResult(false));
     HintUtils.runIfHasType(hint, Retryable.class, retryable -> retryable.setRetry(retry));
+    HintUtils.runIfHasType(
+        hint,
+        DiskFlushNotification.class,
+        (diskFlushNotification) -> {
+          if (diskFlushNotification.isFlushable(envelope.getHeader().getEventId())) {
```

Comment:
tbh I think it's fine to just `markFlushed` and don't wait for the bound transaction if we are rate-limited, but this is also fine as-is now

Reply:
Thanks, I'll change it to always `markFlushed` then.